File: leet_code/go/jewelsAndStones.go
import "strings"

func numJewelsInStones(J string, S string) int {
	jewels := make(map[string]bool)
	arr := strings.Split(J, "")
	for _, a := range arr {
		jewels[a] = true
	}
	count := 0
	arr = strings.Split(S, "")
	for _, a := range arr {
		if _, ok := jewels[a]; ok {
			count++
		}
	}
	return count
}

File: leet_code/ruby/jewels_and_stones.rb
# @param {String} j
# @param {String} s
# @return {Integer}
def num_jewels_in_stones(j, s)
	jewels = Hash.new(false)
	j.each_char { |a| jewels[a] = true }
	count = 0
	s.each_char { |a| count += 1 if jewels[a] }
	count
end